drivers/display/display_sdl.c

```diff
@@ -509,34 +597,57 @@ static int sdl_display_clear(const struct device *dev)
 		return -EINVAL;
 	}
 	LOG_DBG("size: %zu, bgcolor: %hhu", size, bgcolor);
-	memset(disp_data->buf, bgcolor, size);
+
+	K_SPINLOCK(&disp_data->lock) {
```

Comment:
sidenote: calls to memset are atomic in native_sim

Reply:
Done.